components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/DataFlowProcessingEngine.cs

```diff
@@ -159,33 +160,69 @@ public class DataFlowProcessingEngine : IProcessingEngine, IDisposable {
+        public JobDiagnosticInfoModel GetDiagnosticInfo() {
+            var totalSeconds = (DateTime.UtcNow - _diagnosticStart).TotalSeconds;
+            double totalDuration = _diagnosticStart != DateTime.MinValue ? totalSeconds : 0;
+            double chunkSizeAverage = _messageEncoder.AvgMessageSize / (4 * 1024);
+            double sentMessagesPerSec = totalDuration > 0 ? _messageSink.SentMessagesCount / totalDuration : 0;
+            double estimatedMsgChunksPerDay = Math.Ceiling(chunkSizeAverage) * sentMessagesPerSec * 60 * 60 * 24;
+            var diagnosticInfo = new JobDiagnosticInfoModel();
+
+            diagnosticInfo.Id = Name;
+            diagnosticInfo.IngestionDuration = TimeSpan.FromSeconds(totalDuration);
+            diagnosticInfo.IngressDataChanges = _messageTrigger.DataChangesCount;
+            diagnosticInfo.IngressValueChanges = _messageTrigger.ValueChangesCount;
+            diagnosticInfo.IngressBatchBlockBufferSize = _batchDataSetMessageBlock.OutputCount;
+            diagnosticInfo.EncodingBlockInputSize = _encodingBlock.InputCount;
+            diagnosticInfo.EncodingBlockOutputSize = _encodingBlock.OutputCount;
+            diagnosticInfo.EncoderNotificationsProcessed = _messageEncoder.NotificationsProcessedCount;
+            diagnosticInfo.EncoderNotificationsDropped = _messageEncoder.NotificationsDroppedCount;
+            diagnosticInfo.EncoderIoTMessagesProcessed = _messageEncoder.MessagesProcessedCount;
+            diagnosticInfo.EncoderAvgNotificationsMessage = _messageEncoder.AvgNotificationsPerMessage;
+            diagnosticInfo.EncoderAvgIoTMessageBodySize = _messageEncoder.AvgMessageSize;
+            diagnosticInfo.EncoderAvgIoTChunkUsage = chunkSizeAverage;
+            diagnosticInfo.EstimatedIoTChunksPerDay = estimatedMsgChunksPerDay;
+            diagnosticInfo.OutgressBatchBlockBufferSize = _batchNetworkMessageBlock.OutputCount;
+            diagnosticInfo.OutgressInputBufferCount = _sinkBlock.InputCount;
+            diagnosticInfo.OutgressInputBufferDropped = _sinkBlockInputDroppedCount;
+            diagnosticInfo.OutgressIoTMessageCount = _messageSink.SentMessagesCount;
+            diagnosticInfo.ConnectionRetries = _messageTrigger.NumberOfConnectionRetries;
+            diagnosticInfo.OpcEndpointConnected = _messageTrigger.IsConnectionOk;
+            diagnosticInfo.MonitoredOpcNodesSucceededCount = _messageTrigger.NumberOfGoodNodes;
+            diagnosticInfo.MonitoredOpcNodesFailedCount = _messageTrigger.NumberOfBadNodes;
+
+            return diagnosticInfo;
+        }
+
         /// <summary>
         /// Diagnostics timer
         /// </summary>
         /// <param name="state"></param>
         private void DiagnosticsOutputTimer_Elapsed(object state) {
+            var info = GetDiagnosticInfo();
             var totalSeconds = (DateTime.UtcNow - _diagnosticStart).TotalSeconds;
             double totalDuration = _diagnosticStart != DateTime.MinValue ? totalSeconds : 0;
```

Comment:
`totalDuration` can also be taken from `info.IngestionDuration`.

Reply:
done